trustfall_core/src/frontend/error.rs

```diff
@@ -31,6 +31,11 @@ pub enum FrontendError {
     )]
     TagUsedOutsideItsFoldedSubquery(String, String),
 
+    #[error(
+        "It's not allowed to apply \"{0}\" to a property, but it was applied to \"{1}\" property ."
+    )]
+    UnsupportedDirectiveOnProperty(String, String),
```

Comment:
As another option for placing the variant, consider having it next to `UnsupportedEdgeOutput`. That variant is about an unsupported directive on an edge, which nicely parallels this variant's complaint about an unsupported directive on a property.